File: Assignment 12/program2.c
////////////////////////////////////////////////////////////////////////////////
//
//  Required Heder Files
//
////////////////////////////////////////////////////////////////////////////////
#include<stdio.h>       //For Input Output
////////////////////////////////////////////////////////////////////////////////
//
//  Function Name : BOOL chkZero
//  Description   : It is used to chrck whether it contains 0 in it or not 
//  Input         : Intiger  
//  OutPut        : Intiger  
//  Author        : Pratibha Sagar Pawar
//  Date          : 1 Nov 2025
//      
////////////////////////////////////////////////////////////////////////////////

#define TRUE 1
#define FALSE 0

typedef int BOOL;

BOOL ChkZero(int iNo)
{
    int iDigit =0;
    if(iNo<0)
    {
        iNo = -iNo;
    }
    while(iNo>0)
    {
        iDigit = iNo % 10;
        if(iDigit==0)
        {
            return TRUE;
        }
        else
        {
            iNo = iNo/10;
            return FALSE;
        }
    }
}
////////////////////////////////////////////////////////////////////////////////
//
//  Entery point function for the application 
// 
////////////////////////////////////////////////////////////////////////////////
int main()
{
    int iValue= 0;
    BOOL bRet = FALSE;

    printf("Enter number :");
    scanf("%d",&iValue);

    bRet = ChkZero(iValue);
    if(bRet == TRUE)
    {
        printf("It Contains Zero");
    }
    else
    {
        printf("There is no Zero");
    }
    return 0;
}
//  End of main Function
////////////////////////////////////////////////////////////////////////////////
//
//  Input : 725    Output : There is no zero
//  Input : 401    Output : It Contains Zero     
//  Inpur : 212    Output : There is no zero
//
////////////////////////////////////////////////////////////////////////////////

File: Assignment 31/program2.java
////////////////////////////////////////////////////////////////////////////////

import java.util.Scanner;

////////////////////////////////////////////////////////////////////////////////
//
//  Class Name    : class pattern
//  Description   : It is used to Display pattern 
//  Input         : Intiger  
//  OutPut        : Pattern 
//  Author        : Pratibha Sagar Pawar
//  Date          : 22 Nov 2025
//      
////////////////////////////////////////////////////////////////////////////////
class pattern
{
    public void Display(int iRow, int iCol)
    {
        int i = 0;
        int j = 0;
       
            for(i = 1; i<=iRow; i++ )
        {
            for(j = 1; j<=iCol; j++)
            {
                if(j <=iCol - i + 1  )
                {
                    System.out.print("*\t");
                }
                else
                {
                   System.out.print("\t");
                }
        
            }   System.out.println();
        }
        System.out.println();
    }
}
///////////////////////////////////////////////////////////////////////////////
//
//  Entery point method for the application 
// 
////////////////////////////////////////////////////////////////////////////////

class program2
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);

        int iValue1 = 0;
        int iValue2 = 0;

        System.out.println("Enter the number of Rows : ");
        iValue1 = sobj.nextInt();

        System.out.println("Enter the number of columns : ");
        iValue2 = sobj.nextInt();

        pattern pobj = new pattern();
        pobj.Display(iValue1, iValue2);

    }
}
//  End of main method
////////////////////////////////////////////////////////////////////////////////
//
//  Input  : iRow = 4  iCol = 4
//
//  Output :  *    *    *    *
//            *    *    *    
//            *    *        
//            *            
////////////////////////////////////////////////////////////////////////////////


